### ISSUE TITLE
Publish fails from VS, because `project.assets.json` not found

### ISSUE THREAD

**Issue by Marv51** [microsoft/microsoft-ui-xaml#5248]
**Describe the bug**
When I create a new Solution with an empty WinUI 3 Desktop app in VS 16.10.2. The project works both in Debug and Release configuration without error. However, when I use VS to create MSIX packages, the creation fails.

```
1>------ Publish started: Project: App10 (Package), Configuration: Release x86 ------
1>Determining projects to restore...
1>Restored C:\Users\Marvin\source\repos\App10\App10\App10.csproj (in 478 ms).
1>Assets file 'C:\Users\Marvin\source\repos\App10\App10\obj\wappublish\win10-x86\project.assets.json' not found. Run a NuGet package restore to generate this file.
========== Publish: 0 succeeded, 1 failed, 0 skipped ==========
========== Package: 0 succeeded, 2 failed ===========
```

This seems indeed correct, as the file project.assets.json it is looking for is in the path `C:\Users\Marvin\source\repos\App10\App10\obj\wappublish\win-x86\project.assets.json` (Notice the `win-x86` instead of `win10-x86`)

Can I tell VS to look in the right directory?

**Project source**
[App10.zip](https://github.com/microsoft/microsoft-ui-xaml/files/6699936/App10.zip)

**Build bin logs**
[logs.zip](https://github.com/microsoft/microsoft-ui-xaml/files/6699917/logs.zip)

**Version Info**

NuGet package version: 

WinUI 3 - Project Reunion 0.5: 0.5.7

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  | Yes |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2021 Update (19043)        | Yes  |
| October 2020 Update (19042)        |    |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**
This is how I created MSIX packages in VS:
Right click in Solution Explorer on the Packaging project -> Publish -> Create App Packages. I select Sideloading, disable automatic updates, select my certificate for signing, select Release (x86) only. Generate app bundle "Always".

**Comment by StephenLPeters** [comment]
@alwu-msft is VS 16.10.2 working?

@ranjeshj I think you had a bug you were encoutering with this VS version as well, recognize this?

**Comment by fabiant3** [comment]
This is a known issue and it was fixed in 16.10.2.

**Comment by edwardskrod** [comment]
This issue was actually fixed in 16.11, not in 16.10.2.

**Comment by edwardskrod** [comment]
We are going to try to take this for 16.10.4.  I will keep you posted.

**Comment by Marv51** [comment]
I can confirm that this is fixed in 16.10.4
